types/shelljs/index.d.ts

```diff
@@ -965,6 +966,51 @@ export type ShellString = string & ShellReturnValue;
 
 export type ShellArray = string[] & ShellReturnValue;
 
+export const ShellString: ShellStringFunction;
+
+export interface ShellStringFunction {
+	/**
+	 * Wraps a string (or array) value. This has all the string (or array) methods,
+	 * but also exposes extra methods: `.to()`, `.toEnd()`, and all the pipe-able
+	 * methods (ex. `.cat()`, `.grep()`, etc.).
+	 *
+	 * This can be easily converted into a string by calling `.toString()`.
+	 *
+	 * This type also exposes the corresponding command's stdout, stderr, and return status
+	 * code via the `.stdout` (string), `.stderr` (string), and `.code` (number) properties
+	 * respectively.
+	 *
+	 * Construct signature allows for:
+	 *
+	 * var foo = new ShellString('hello world');
+	 *
+	 * as per example in shelljs docs:
+	 * https://github.com/shelljs/shelljs#shellstringstr
+	 *
+	 * @param value 	The string value to wrap.
+	 * @return				A string-like object with special methods.
+	 */
+	new(value: string): ShellString;
+	new(value: string[]): ShellArray;
+
+	/**
+	 * Wraps a string (or array) value. This has all the string (or array) methods,
+	 * but also exposes extra methods: `.to()`, `.toEnd()`, and all the pipe-able
+	 * methods (ex. `.cat()`, `.grep()`, etc.).
+	 *
+	 * This can be easily converted into a string by calling `.toString()`.
+	 *
+	 * This type also exposes the corresponding command's stdout, stderr, and return status
+	 * code via the `.stdout` (string), `.stderr` (string), and `.code` (number) properties
+	 * respectively.
+	 *
+	 * @param value 	The string value to wrap.
+	 * @return				A string-like object with special methods.
+	 */
+	(value: string): ShellString;
+	(value: string[]): ShellArray;
+}
```

Comment:
I&nbsp;prefer putting the&nbsp;<code>export&nbsp;const</code> declaration after the&nbsp;interface declaration, to&nbsp;match&nbsp;the&nbsp;rest&nbsp;of&nbsp;this type&nbsp;definition and&nbsp;[TypeScript’s&nbsp;standard library](https://github.com/Microsoft/TypeScript/blob/42302705f6d4266778324d8282ed743ffa305d33/lib/lib.es5.d.ts#L517-L527):
```suggestion
}

export const ShellString: ShellStringConstructor;
```